src/lua/compat.lua

```diff
@@ -0,0 +1,236 @@
+Available commands:
+
+    candidates()                    -- list all unselected options
+    get_setup()                     -- get Lua command that sets up different compat with same options as current
+    help()                          -- show this help
+    reset()                         -- set all options to default
+    add_options()                   -- add new options by providing name, option table and postaction for each one
+    restore{'option_name'}          -- set to default specified options
+    {option_name = true}            -- set listed options to desired values (true, false, 'old', 'new', 'default')
+    option_name                     -- list option info
+    option_name = true              -- set desired value to option, could be true, false, 'old', 'new', 'default'
+]]
+
+-- Returns table with combined info on option from `options` and `cfg`.
+local function serialize_policy(key, policy)
+    assert(key    ~= nil)
+    assert(policy ~= nil)
+    local result = table.deepcopy(policy)
+    result.value = cfg[key].value
+    return result
+end
+
+-- FIXME should it list frozen options?
+-- Returns info on all unfrozen options from `options`.
+local function serialize_compat()
+    local result = { }
+    for key, val in pairs(options) do
+        result[key] = serialize_policy(key, val)
+    end
+    return result
+end
+
+-- Checks options correctness and sets `option.default` to boolean value if needed.
+local function verify_option(name, option)
+    if type(name) ~= 'string' then
+        error(('Option name must be a string (%s provided)'):format(type(name)))
+    end
+    if option.default == 'new' then
+        option.default = option.new
+    elseif option.default == 'old' then
+        option.default = option.old
+    end
+    for p, t in pairs(options_format) do
+        if type(option[p]) ~= t then
+            error(('Invalid option table for %s, bad %s (%s is expected)'):format(name, p, t))
+        end
+    end
+    if option.frozen and not option.default == option.new then
+        error(('Frozen option %s default is wrong'):format(name))
+    end
+    if option.new == option.old then
+        error(('In option %s old and new contain the same value'):format(name))
+    end
+end
+
+-- Checks if operation is valid, sets value to an option and runs postaction.
+local function set_option(name, val)
+    local option = options[name]
+    if not option then
+        error(('Invalid option %s'):format(name))
+    end
+    if val == 'new' then
+        val = option.new
+    end
+    if val == 'old' then
+        val = option.old
+    end
+    local default = false;
+    if val == 'default' then
+        val = option.default
+        default = true
+    end
+    if type(val) ~= 'boolean' then
+        error(('Invalid argument %s'):format(val))
+    end
+    local log = require('log')          -- Log can't be required at the beginning, as it isn't initialized by then.
```

Comment:
Put all `require` to the top of file.

Reply:
`log` can't be required at the beginning, as it isn't initialized by then.